pkg/rook/purge_osd.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2023 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rook
+
+import (
+	ctx "context"
+	"fmt"
+	"log"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/mons"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func PurgeOsd(context *k8sutil.Context, operatorNamespace, clusterNamespace, osdId, flag string) string {
+
+	monCm, err := context.Clientset.CoreV1().ConfigMaps(clusterNamespace).Get(ctx.TODO(), mons.MonConfigMap, v1.GetOptions{})
+	if err != nil {
+		log.Fatalf("failed to get mon configmap %s %v", mons.MonConfigMap, err)
+	}
+
+	monEndPoint := monCm.Data["data"]
+	cephArgs := []string{
+		"auth", "print-key", "client.admin",
+	}
+	adminKey := k8sutil.RunCommandInOperatorPod(context, "ceph", cephArgs, operatorNamespace, clusterNamespace)
+
+	cmd := "/bin/sh"
+	args := []string{
+		"-c",
```

Comment:
Doesn't the `RunCommandInOperatorPod()` method already take care of the `/bin/sh` and `-c` arg? Or it would be nice if it did, so this caller only has to worry about its own command and arguments

Reply:
I don't we can since we run ceph/rbd commands too and if we want to take care of `/bin/sh -c` it will fail since it wants other commands in quotes.
like `/bin/sh -c  "ceph -s"` not sure who we can do  it here

Reply:
How about a helper such as `RunShellCommandInOperatorPod()`? It might also be nice to take the env vars as a separate parameter. It would just be nice to push the details of command execution into the `exec` package.